### PR TITLE
Update dotnet-getdocument to support minimal APIs

### PR DESCRIPTION
**Pull request by captainsafia**
## Description

This PR update the `dotnet getdocument` tool to react to changes in hosting that were shipped in .NET 6.

The hosting changes were incongruent with the strategy that the `getdocument` tool used to extract the active ServiceProvider from the application assembly. 

Fixes https://github.com/dotnet/aspnetcore/issues/43391

## Customer Impact

Without this change, OpenAPI document generation in the CLI and client generation in Visual Studio would fail to produce the necessary schemas and code for minimal API endpoints in user applications.

## Regression?

- [ ] Yes
- [X] No

## Risk

- [ ] High
- [ ] Medium
- [X] Low

Low because:
- Change is limited to `getdocument` tool
- Change follows pattern used to react to hosting changes in other apps that examine host

## Verification

- [X] Manual (required)
- [ ] Automated

## Packaging changes reviewed?

- [ ] Yes
- [ ] No
- [X] N/A

### CODE REVIEW

File: src/Tools/GetDocumentInsider/src/Commands/GetDocumentCommand.cs [hunk 1]
```diff
@@ -5,7 +5,7 @@
 using System.IO;
 using System.Linq;
 using System.Reflection;
-#if NETCOREAPP2_1
+#if NETCOREAPP2_1 || NET7_0_OR_GREATER
```
Comment (BrennanConroy) [inline src/Tools/GetDocumentInsider/src/Commands/GetDocumentCommand.cs:8]: Is it really anything netcoreapp2.1 and above? I think `NETCOREAPP` covers that.

File: src/Tools/GetDocumentInsider/src/Commands/GetDocumentCommandWorker.cs [hunk 2]
```diff
@@ -53,6 +60,88 @@ public int Process()
...
+            {
+                services.AddSingleton<IServer, NoopServer>();
+                services.AddSingleton<IHostLifetime, NoopHostLifetime>();
+            });
+        }
+
+        // Register a TCS to be invoked when the entrypoint (aka Program.Main)
+        // has finished running. For minimal APIs, this means that all app.X
+        // calls about the host has been built have been executed.
+        var waitForStartTcs = new TaskCompletionSource<object>();
```
Comment (BrennanConroy) [inline src/Tools/GetDocumentInsider/src/Commands/GetDocumentCommandWorker.cs:78]: `TaskCreatingOptions.RunContinuationsAsynchronously`

File: src/Tools/GetDocumentInsider/src/Commands/GetDocumentCommandWorker.cs [hunk 3]
```diff
@@ -53,6 +60,88 @@ public int Process()
...
+                    typeof(IServiceProvider),
+                    HostFactoryResolver.BuildWebHost,
+                    HostFactoryResolver.CreateHostBuilder,
+                    HostFactoryResolver.CreateWebHostBuilder,
+                    entryPointType));
+
+                return 5;
+            }
+
+            // Wait for th application to start to ensure that all configurations
```
Comment (BrennanConroy) [inline src/Tools/GetDocumentInsider/src/Commands/GetDocumentCommandWorker.cs:126]:
```suggestion
            // Wait for the application to start to ensure that all configurations
```

File: src/Tools/dotnet-getdocument/src/Commands/InvokeCommand.cs [hunk 4]
```diff
@@ -79,9 +79,16 @@ protected override int Execute()
                             projectName,
                             targetFramework.Version));
                     }
+                    else if (targetFramework.Version > new Version(7, 0))
+                    {
+                        toolsDirectory = Path.Combine(thisPath, "net7.0");
```
Comment (martincostello) [inline src/Tools/dotnet-getdocument/src/Commands/InvokeCommand.cs:84]: Should this format from the property value so it doesn't need changing if the TFM gets bumped in a future version?

```suggestion
                        toolsDirectory = Path.Combine(thisPath, $"net{targetFramework.ToString(2)}");
```

File: src/Tools/GetDocumentInsider/src/Commands/GetDocumentCommandWorker.cs [hunk 5]
```diff
@@ -53,6 +60,88 @@ public int Process()
...
+
+        if (factory == null)
+        {
+            _reporter.WriteError(Resources.FormatMethodsNotFound(
+                HostFactoryResolver.BuildWebHost,
+                HostFactoryResolver.CreateHostBuilder,
+                HostFactoryResolver.CreateWebHostBuilder,
+                entryPointType));
+
+            return 4;
```
Comment (dougbu) [inline src/Tools/GetDocumentInsider/src/Commands/GetDocumentCommandWorker.cs:107]: Start the new `return` integers at `8`. I already used `return 0` through `return 7` in `GetDocumentInsider`.

File: src/Tools/dotnet-getdocument/src/Commands/InvokeCommand.cs [hunk 6]
```diff
@@ -79,9 +79,16 @@ protected override int Execute()
                             projectName,
                             targetFramework.Version));
                     }
+                    else if (targetFramework.Version > new Version(7, 0))
```
Comment (dougbu) [inline src/Tools/dotnet-getdocument/src/Commands/InvokeCommand.cs:82]:
```suggestion
                    else if (targetFramework.Version >= new Version(7, 0))
```

**Review by dougbu — APPROVED** [review]
Approving w/ the assumption that you address the comments below. I don't need to check again 😀

File: src/Tools/GetDocumentInsider/src/Commands/GetDocumentCommandWorker.cs (hunk 5 above)
Comment (captainsafia) [reply src/Tools/GetDocumentInsider/src/Commands/GetDocumentCommandWorker.cs:107]: Can you clarify what you mean? This codepath is inside `GetDocumentInsider` and the return codes used here match the ones used in the code that is now in the `else` of this `#ifdef`.

Comment (dougbu) [reply src/Tools/GetDocumentInsider/src/Commands/GetDocumentCommandWorker.cs:107]: The idea was to make the `return` codes completely unique, making it dead easy to find what went wrong.

Comment (captainsafia) [reply src/Tools/GetDocumentInsider/src/Commands/GetDocumentCommandWorker.cs]: Ah, got it. I've updated these.

**Comment by captainsafia** [comment]
@Pilchie Can I get RC2 approval for this?

**Comment by Pilchie** [comment]
Approved for .NET 7 RC2.

**Pull request merged**